src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -141,7 +145,7 @@ public boolean deleteFile(FireStoreFile fireStoreFile) {
     BillingProfileModel profileModel =
```

Comment:
Another refactoring opportunity.  You could implement the logic of deleting the parent folder inside the `BlobCrl`. Either as a new method, `BlobCrl.deleteBlobAndParentFolder`, or as an overloaded method like this: `BlobCrl.deleteBlob(String blobName, boolean deleteParentFolder)`

Reply:
I was torn about that...not sure if it puts too much on a lower level library